deny.toml

```diff
@@ -0,0 +1,61 @@
+[advisories]
+db-path = "~/.cargo/advisory-db"
+db-urls = ["https://github.com/rustsec/advisory-db"]
+vulnerability = "deny"
+unmaintained = "deny"
+yanked = "deny"
+notice = "deny"
+ignore = [
+    "RUSTSEC-2020-0016", # net2 deprecated - https://github.com/deprecrated/net2-rs/commit/3350e3819adf151709047e93f25583a5df681091
+    "RUSTSEC-2020-0056", # stdweb unmaintained - https://github.com/koute/stdweb/issues/403
+    "RUSTSEC-2021-0047", # security issue - https://github.com/gnzlbg/slice_deque/issues/90
+]
+
+[licenses]
+unlicensed = "allow"
+copyleft = "allow"
```

Comment:
Are we sure we want to allow `copyleft` licenses?
IIUC
1) this prevents using bevy in closed source projects 
2) if we use a crate which has a copyleft license, bevy must also adopt that copyleft license (e.g. GPL)

I might have misread the legalese with copyleft, so please correct me if I'm wrong 😄

Reply:
The config used is not "do we want to or not", it's what needed for the check to pass, so it's the current status. 

wgpu is licensed under MPL-2.0 which is considered copyleft

Reply:
yeah we don't want GPL or LGPL, so allowing only MPL seems right

Reply:
Yeah MPL isn't problematic in the same way as GPL because it isn't "viral". It can be used in a wider context without the need to release that context under the same license. It _does_ require that the license be included with binaries, which isn't great, but its also worth pointing out that this is also (generally considered to be) true for MIT. There isn't a practical way to avoid open source software that has this requirement because almost all popular licenses have this requirement. 

In most cases, people shouldn't need to care. Teenagers publishing their Godot games on itch.io aren't getting sued for damages by FreeType or ENet. To my knowledge, even professionals aren't. However professionals have more to lose and generally choose to err on the side of caution and comply by either embedding licenses in their game credits or including them in installations. Like Godot, we should provide a [license compliance guide](https://docs.godotengine.org/en/stable/tutorials/legal/complying_with_licenses.html) for people who want to be 100% safe. Ideally we have an auto-generated uber-license file that people can drop somewhere / embed / link to.

We could also consider getting "cute" here by doing something like `include_bytes!("uber_license_file")` by default (with a way to opt out for people who want slimmer binaries and have other compliance plans), but thats not really in the spirit of these licenses and might not even hold up in court.

It would be nice to also get a verbal assurance that @kvark won't litigate teenagers publishing their Bevy games on itch.io without the wgpu license file. Of course ideally they add a provision that binaries don't need to include the license. We should consider doing the same for Bevy as I have no interest in forcing that on anyone. But again, theres no practical way of avoiding this problem entirely for something as big as an engine. Eventually we will hit something foundational that we can't reasonably build in house (ex: something openssl).

Reply:
I have better things to do in life than chasing them teenagers :) Please count it as my verbal assurance.